src/credentials/CHIPCert.h

```diff
@@ -641,6 +641,81 @@ CHIP_ERROR ConvertX509CertToChipCert(const uint8_t * x509Cert, uint32_t x509Cert
 CHIP_ERROR ConvertChipCertToX509Cert(const uint8_t * chipCert, uint32_t chipCertLen, uint8_t * x509CertBuf,
                                      uint32_t x509CertBufSize, uint32_t & x509CertLen);
 
+/**
+ * @brief Generate a standard X.509 DER encoded certificate using provided CHIP certificate and signing key
+ *
+ * @param chipCert        Buffer containing CHIP certificate.
+ * @param chipCertLen     The length of the CHIP certificate.
+ * @param keypair         The certificate signing key
+ * @param x509CertBuf     Buffer to store signed certificate in X.509 DER format.
+ * @param x509CertBufSize The size of the buffer to store converted certificate.
+ * @param x509CertLen     The length of the converted certificate.
+ *
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ **/
+CHIP_ERROR GenerateSignedX509CertFromChipCert(const uint8_t * chipCert, uint32_t chipCertLen, Crypto::P256Keypair & keypair,
+                                              uint8_t * x509CertBuf, uint32_t x509CertBufSize, uint32_t & x509CertLen);
+
+struct X509CertRequestParams
+{
+    int64_t mSerialNumber;
```

Comment:
nit: should we have a 'm' prefix for structs? I am generally used to have previx/suffix for classes but not inside structs. Thoughts?

The argument for not having prefixes: structs members are public so when I see `foo.mBar = 1` I immediately think that this looks like a bad encapsulation (class exposing data members) but this is fine for structs.